src/open_options.rs

```diff
@@ -71,9 +74,42 @@ impl OpenOptions {
             truncate: false,
             create: false,
             create_new: false,
+            buffer_size: 0,
+            replication: 0,
+            blocksize: 0,
         }
     }
 
+    /// Sets size of buffer for read/write.
+    ///
+    /// Pass `0` if you want to use the default configured values.
+    ///
+    /// `0` by default.
+    pub fn with_buffer_size(&mut self, buffer_size: c_int) -> &mut Self {
+        self.buffer_size = buffer_size;
+        self
+    }
+
+    /// Sets block replication.
+    ///
+    /// Pass `0` if you want to use the default configured values.
+    ///
+    /// `0` by default.
+    pub fn with_replication(&mut self, replication: c_short) -> &mut Self {
+        self.replication = replication;
+        self
+    }
+
+    /// Sets size of block.
+    ///
+    /// Pass `0` if you want to use the default configured values.
+    ///
+    /// `0` by default.
+    pub fn with_blocksize(&mut self, blocksize: i32) -> &mut Self {
```

Comment:
What will happen if `blocksize < 0`? What doesn `0` mean, the blocksize in hdfs config?

Reply:
Strictly speaking, the type should be `u31` according to [the valid values in the "Block Size" section](https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-hdfs/WebHDFS.html). `0` is [defined](https://github.com/avaranovich/libhdfs/blob/79090c2ef7c86cc8259ca2c7a6ec716292daca60/native/include/hdfs.h#L273) to be a default in C API. Using the web API as a cross-reference, `0` means "Specified in the configuration".

Since there isn't a `u31` available, using `u16` seems too restrictive while `u32` straight up welcoming invalid values, I suggest we just do it in the C API way and have users informed by `Error::last_os_error()` instead.

Reply:
... if users pass in any invalid values.

Reply:
We can verify those things while calling `open`.